src/main/java/org/reactome/addlinks/AddLinks.java

```diff
@@ -193,96 +197,88 @@ public void doAddLinks() throws Exception
 		this.reportsAfterAddLinks(xrefReporter, duplicateIdentifierReporter, preAddLinksReport);
 		logger.info("Purging unused ReferenceDatabse objects.");
 		this.purgeUnusedRefDBs();
-
-		logger.info("Now checking links.");
-
-		String linksReport = this.checkLinks();
-		String diffReportName = LINK_CHECK_REPORTS_PATH + "/linkCheckSummaryReport" + DateTimeFormatter.ofPattern(DATE_PATTERN_FOR_FILENAMES).format(LocalDateTime.now()) + ".tsv";
-		Files.write(Paths.get(diffReportName), linksReport.getBytes() );
-
 		logger.info("Process complete.");
 	}
 
 	/**
-	 * Checks the links to external resources. Returns a
-	 * @return
+	 * @param linkCheckReportLines
+	 * @param linkCheckManager
+	 * @param refDBInst
+	 * @param reportLine
+	 * @param numLinkOK
+	 * @param numLinkNotOK
 	 */
-	private String checkLinks()
-	{
-		StringBuilder linkCheckReportLines = new StringBuilder("RefDBName\tNumOK\tNumNotOK\n");
-		// Now, check the links that were created to ensure that they are all valid.
-		LinkCheckManager linkCheckManager = new LinkCheckManager();
-		linkCheckManager.setDbAdaptor(this.dbAdapter);
-		// Filter by references database name.
-		for (GKInstance refDBInst : LinksToCheckCache.getCache().keySet() )
+	private String checkLinksForRefDB(LinkCheckManager linkCheckManager, GKInstance refDBInst)
+	{
+		 int numLinkOK = 0;
+		 int numLinkNotOK = 0;
+		logger.info("Link-checking for database: {}", refDBInst.getDisplayName());
+		StringBuilder reportLine = new StringBuilder();
+		if (!refDBInst.getDisplayName().toUpperCase().contains(ENSEMBL))
+		{
+			reportLine.append(refDBInst.getDisplayName()).append("\t");
+		}
```

Comment:
Why does the report have a tab appended if the reference db's display name doesn't contain "ENSEMBL"?

Reply:
ENSEMBL names require special handling.

IF the ReferenceDatabase is _NOT_ ENSEMBL, append the reference database name and a tab to the current report line. The tab is because the output is a TSV file, so after the value of the report name, there must be a tab.
ELSE (the ReferenceDatabase _is_ ENSEMBL), find a longer name because ENSEMBL ReferencesDatabases have multiple names. All of them have the name "ENSEMBL", but for the report, we want a name that's species-specific. The easiest way to do that is to find a name that is longer than the the string "ENSEMBL". Append _that_ name to the report line, with a tab.